magento/magento2
# Fix cache validation support for Redis Enterprise and Clusters

### Description (*)
This changes, slightly adjust regexp for cache ID's/tags specifically for Redis cache interface. In Redis it's possible to control data flow in cluster mode by using tags in curly brackets {}, so it's allowed to use key names with them and allows to have more control on data flows. Fixes bug with duplicated keys on different shards.

### Manual testing scenarios (*)
1. Create an environment with Redis Enterprise cluster with at least 1 node and DB with 2 shards.
2. Setup cache configuration in `env.php` to write into Redis, `id_prefix` value should be like `{...}`.
3. Login into admin panel now, shouldn't throw an exception. (To reproduce exception just remove curly brackets from prefix).

### Contribution checklist (*)
